app/src/androidTest/java/co/nimblehq/compose/crypto/ui/screen/HomeScreenUITest.kt

```diff
@@ -0,0 +1,202 @@
+
+    private val todayProfitLabel: String
+        get() = composeAndroidTestRule.activity.getString(R.string.portfolio_card_today_profit_label)
+
+    private val errorGeneric: String
+        get() = composeAndroidTestRule.activity.getString(R.string.error_generic)
+
+    private val expectedPriceChange: String
+        get() = composeAndroidTestRule.activity.getString(
+            R.string.coin_profit_percent,
+            MockUtil.trendingCoins.first().priceChangePercentage24hInCurrency.toFormattedString()
+        )
+
+    private val mockGetMyCoinsUseCase = mockk<GetMyCoinsUseCase>()
+    private val mockGetTrendingCoinsUseCase = mockk<GetTrendingCoinsUseCase>()
+
+    private lateinit var viewModel: HomeViewModel
+
+    private var appDestination: AppDestination? = null
+
+    @Before
+    fun setUp() {
+        composeAndroidTestRule.activity.setContent {
+            HomeScreen(
+                viewModel = viewModel,
+                navigator = { destination -> appDestination = destination }
+            )
+        }
+
+        every { mockGetMyCoinsUseCase.execute(any()) } returns flowOf(MockUtil.myCoins)
+        every { mockGetTrendingCoinsUseCase.execute(any()) } returns flowOf(MockUtil.trendingCoins)
+    }
+
+    @Test
+    fun hen_entering_HomeScreen__it_renders_the_PortfolioCard_properly() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(testTag = TestTagHomeTitle).assertTextEquals(homeTitle)
+            onNodeWithTag(testTag = TestTagTotalCoinsLabel).assertTextEquals(totalCoinsLabel)
+            onNodeWithTag(testTag = TestTagTodayCoinProfitLabel).assertTextEquals(todayProfitLabel)
+            onNodeWithTag(testTag = TestTagCardTotalCoins).assertTextEquals("$7,273,291")
+            onNodeWithTag(testTag = TestTagCardTodayProfit).assertTextEquals("$193,280")
+        }
+    }
+
+    @Test
+    fun when_loading_MyCoins__it_renders_the_LoadingProgress_properly() {
+        every { mockGetMyCoinsUseCase.execute(any()) } returns flow { delay(500) }
+
+        initViewModel()
+
+        composeAndroidTestRule.onNodeWithTag(testTag = TestTagCoinsLoader).assertIsDisplayed()
+    }
+
+    @Test
+    fun when_loading_TrendingCoins__it_renders_the_LoadingProgress_properly() {
+        every { mockGetTrendingCoinsUseCase.execute(any()) } returns flow { delay(500) }
+
+        initViewModel()
+
+        composeAndroidTestRule.onNodeWithTag(testTag = TestTagCoinsLoader).assertIsDisplayed()
+    }
+
+    @Test
+    fun when_entering_HomeScreen_and_loading_MyCoins_successfully__it_renders_the_UI_properly() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            with(MockUtil.myCoins.first()) {
+                onAllNodesWithTag(
+                    testTag = TestTagCoinItemSymbol,
+                    useUnmergedTree = true
+                ).onFirst().assertTextEquals(symbol.uppercase())
```

Comment:
Please consider asserting with text value. It is not a good practice to include test code inside the production code. We can add `testTag` for components like `Image` for testing but for `Buttons` and `Text` components, we should assert with mocked text value. 🙏 
```suggestion
                onAllNodesWithText(symbol.uppercase()).onFirst().assertIsDisplayed()
```

Reply:
> Please consider asserting with text value

Good point @kaungkhantsoe 👍 and I agreed we should assert by `text` value if we can do and I will update It 🚀 

> It is not a good practice to include test code inside the production code.

And regarding this, I think it's not bad to include it inside the production code. As you said 

> We can add testTag for components like Image